test/syscalls/linux/pty.cc

```diff
@@ -1640,6 +1640,57 @@ TEST_F(JobControlTest, DISABLED_SetForegroundProcessGroup) {
   ASSERT_NO_ERRNO(res);
 }
 
+// This test verify if a SIGTTOU signal is sent to the calling process'group
+// when tcsetpgrp is called by a background process
+TEST_F(JobControlTest, SetForegroundProcessGroupSIGTTOUBackground) {
+  auto res = RunInChild([=]() {
+    setsid();
+    TEST_PCHECK(!ioctl(replica_.get(), TIOCSCTTY, 0));
+    pid_t grandchild = fork();
+    if (!grandchild) {
+        //assign a different pgid to the child so it will result as
```

Comment:
Nit: for consistency, comments start with a space and are full sentences with capitals like:
```cpp
// Assign a different pgid to the child so it will result as
// a background process.
```
This applies to below comments too.